pkg/operator/mon/mon.go

```diff
@@ -76,6 +76,8 @@ type Cluster struct {
 	monPodRetryInterval time.Duration
 	monPodTimeout       time.Duration
 	monTimeoutList      map[string]time.Time
+	HostNetwork         bool
+	assignment          map[string]v1.Node
```

Comment:
There is a scenario where the operator gets restarted after the cluster is running. For example, you can delete rook-operator.yaml, then create it again. The operator should recognize the running cluster and continue running. In that case, you would need to initialize the `assignment` map from existing mons. Is that handled? It likely should be done in `initClusterInfo()`

Reply:
I added a function and a test that should cover exactly the case when the operator restarts.